controllers/cluster_controller_test.go

```diff
@@ -74,40 +78,61 @@ func newVsphereClusterReconcilerTest(t *testing.T, objs ...runtime.Object) *vsph
 	}
 }
 
-func TestClusterReconcilerSkipManagement(t *testing.T) {
-	secret := createSecret()
-	cluster := createCluster()
-	datacenterConfig := createDataCenter(cluster)
-	bundle := createBundle(cluster)
-	machineConfigCP := createCPMachineConfig()
-	machineConfigWN := createWNMachineConfig()
+func TestClusterReconcilerReconcileSelfManagedCluster(t *testing.T) {
+	g := NewWithT(t)
+	ctx := context.Background()
+
+	selfManagedCluster := &anywherev1.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "my-management-cluster",
+		},
+		Spec: anywherev1.ClusterSpec{
+			BundlesRef: &anywherev1.BundlesRef{
+				Name: "my-bundles-ref",
+			},
+		},
+	}
 
-	objs := []runtime.Object{cluster, datacenterConfig, secret, bundle, machineConfigCP, machineConfigWN}
+	log := test.NewNullLogger()
+	controller := gomock.NewController(t)
+	providerReconciler := mocks.NewMockProviderClusterReconciler(controller)
+	registry := newRegistryMock(providerReconciler)
+	c := fake.NewClientBuilder().WithRuntimeObjects(selfManagedCluster).Build()
 
-	tt := newVsphereClusterReconcilerTest(t, objs...)
+	providerReconciler.EXPECT().ReconcileWorkerNodes(ctx, log, sameName(selfManagedCluster))
 
-	req := clusterRequest(cluster)
+	r := controllers.NewClusterReconciler(c, log, registry)
+	result, err := r.Reconcile(ctx, clusterRequest(selfManagedCluster))
+	g.Expect(err).ToNot(HaveOccurred())
+	g.Expect(result).To(Equal(ctrl.Result{}))
```

Comment:
Can we add a test with assertions to show that CP was not reconciled & workers were updated?